src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,42 @@
 package core.basesyntax.service;
 
+import core.basesyntax.InvalidUserDataException;
 import core.basesyntax.dao.StorageDao;
-import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+    private StorageDao storageDao;
+
+    public RegistrationServiceImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new InvalidUserDataException("Login cannot be null.");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidUserDataException("Password cannot be null.");
+        }
+        if (user.getAge() == null) {
+            throw new InvalidUserDataException("Age cannot be null.");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserDataException("User with this login already exists.");
```

Comment:
include login in the exception message, always try to make them as informative as possible